src/document/mod.rs

```diff
@@ -46,14 +40,46 @@ impl ExplicitGrant {
     }
 }
 
+/// Parameters that can be provided when encrypting a new document.
+///
+/// If no document ID is provided, one will be generated for it. If no document name is provided, the document's
+/// name will be left empty.
```

Comment:
```suggestion
/// name will be left empty. Document IDs must be unique per segment.
```

Reply:
The uniqueness is called out in the functions to create a DocumentEncryptOpts, so I would lean toward not including it here. Do you think it's needed in both places?
Or I guess I could also remove it from the functions and _only_ have it here?

Reply:
I'm fine not including it here, but I'd update the line in the other places to say more than just "Unique ID". Having clarity of what we mean by "unique" would be good I think since it only has to be unique within the segment.

Reply:
I decided to have it in both places, since I agree that it makes it more clear